imqsauth/http.go

```diff
@@ -435,9 +435,24 @@ func httpSendPermitsJson(central *ImqsCentral, users []authaus.AuthUser, ident2p
 	httpSendJson(w, jresponse)
 }
 
-func httpSendUserObjectsJson(central *ImqsCentral, users []authaus.AuthUser, ident2perm map[authaus.UserId]*authaus.Permit, w http.ResponseWriter) {
+func httpSendUserObjectsJson(central *ImqsCentral, users []authaus.AuthUser, ident2perm map[authaus.UserId]*authaus.Permit, w http.ResponseWriter, withPerm authaus.PermissionU16) {
```

Comment:
I think we can reasonably split this into two or three functions:

1. `createUserObjectsJson`, which does everything that the old function did, except for sending the result. In addition, add a GroupIDs member to the JSON response - so that we have both the group IDs and group names in the resulting JSON.
2. `filterUserObjectsByPermission`, which does the filtering that you've added here. It can operate on the JSON-ready data structure.
3. The necessary functions which send the HTTP responses.

Reply:
Thanks, will do